utils/main_test.go

```diff
@@ -37,76 +56,91 @@ func TestContainsString(t *testing.T) {
 }
 
 func TestGetUserKey(t *testing.T) {
+	randKey := mkRandRSAPubKey(t)
+
 	table := []struct {
 		key    string
 		user   string
 		keyVal string
 		haserr bool
 	}{
-		{`usera:ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`,
-			"usera", `ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, false},
+		{`usera:ssh-rsa [KEY] google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`,
+			"usera", `ssh-rsa [KEY] google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, false},
+		{`usera:restrict,pty ssh-rsa [KEY] google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`,
+			"usera", `restrict,pty ssh-rsa [KEY] google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, false},
 		{"    ", "", "", true},
-		{"ssh-rsa AAAA1234", "", "", true},
-		{":ssh-rsa AAAA1234", "", "", true},
+		{"ssh-rsa [KEY]", "", "", true},
+		{":ssh-rsa [KEY]", "", "", true},
 		{"userb:", "", "", true},
-		{"userc:ssh-rsa AAAA1234 info text", "userc", "ssh-rsa AAAA1234 info text", false},
+		{"userc:ssh-rsa [KEY] info text", "userc", "ssh-rsa [KEY] info text", false},
 	}
 
 	for _, tt := range table {
-		u, k, err := GetUserKey(tt.key)
+		currentKey := strings.Replace(tt.key, "[KEY]", randKey, 1)
+		currentKeyVal := strings.Replace(tt.keyVal, "[KEY]", randKey, 1)
+		u, k, err := GetUserKey(currentKey)
 		e := err != nil
-		if u != tt.user || k != tt.keyVal || e != tt.haserr {
-			t.Errorf("GetUserKey(%s) incorrect return: got user: %s, key: %s, error: %v - want user: %s, key: %s, error: %v", tt.key, u, k, e, tt.user, tt.keyVal, tt.haserr)
+		if u != tt.user || k != currentKeyVal || e != tt.haserr {
+			t.Errorf("GetUserKey(%s) incorrect return: got user: %s, key: %s, error: %v - want user: %s, key: %s, error: %v", currentKey, u, k, e, tt.user, currentKeyVal, tt.haserr)
 		}
 	}
 }
 
 func TestValidateUserKey(t *testing.T) {
+	randKey := mkRandRSAPubKey(t)
+
 	table := []struct {
 		user   string
 		key    string
 		haserr bool
 	}{
-		{"usera", `ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, false},
-		{"user a", `ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, true},
-		{"usera", `ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"2021-04-23T12:34:56+0000"}`, true},
-		{"usera", `ssh-rsa AAAA1234 google-ssh {"userName":"usera@example.com","expireOn":"Apri 4, 2056"}`, true},
-		{"usera", `ssh-rsa AAAA1234 google-ssh`, true},
-		{"usera", `ssh-rsa AAAA1234 test info`, false},
+		{"usera", `ssh-rsa [KEY] google-ssh {"userName":"usera@example.com","expireOn":"2095-04-23T12:34:56+0000"}`, false},
```

Comment:
Can you please group these test cases by the prefix up to the key token?

Reply:
Done